File: src/capture.py
import cv2

def initialize_capture(source, path=None):
    if source == 'webcam':
        return cv2.VideoCapture(0)
    elif source in ['image', 'video']:
        if not path:
            raise ValueError("You must provide a --path to the image or video file.")
        return cv2.VideoCapture(path)
    else:
        raise ValueError("Invalid source type provided.")

File: src/capture.js
function initializeCapture(source, path = null) {
    if (source === 'webcam') {
        return new cv.VideoCapture(0);  // For the default webcam
    } else if (source === 'image' || source === 'video') {
        if (!path) {
            throw new Error("You must provide a --path to the image or video file.");
        }
        return new cv.VideoCapture(path);  // Open image or video file
    } else {
        throw new Error("Invalid source type provided.");
    }
}

